pkg/network/setup/netstat.go

```diff
@@ -320,10 +321,15 @@ func sriovIfacesStatusFromDomainHostDevices(hostDevices []api.HostDevice, vmiIfa
 	return vmiStatusIfaces
 }
 
-func ifacesStatusFromGuestAgent(vmiIfacesStatus []v1.VirtualMachineInstanceNetworkInterface, guestAgentInterfaces []api.InterfaceStatus) []v1.VirtualMachineInstanceNetworkInterface {
+func ifacesStatusFromGuestAgent(
+	vmiIfacesStatus []v1.VirtualMachineInstanceNetworkInterface,
+	guestAgentInterfaces []api.InterfaceStatus,
+	vmiInterfacesSpecByName map[string]v1.Interface,
+) []v1.VirtualMachineInstanceNetworkInterface {
 	for _, guestAgentInterface := range guestAgentInterfaces {
 		if vmiIfaceStatus := netvmispec.LookupInterfaceStatusByMac(vmiIfacesStatus, guestAgentInterface.Mac); vmiIfaceStatus != nil {
-			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface)
+			vmiIfaceSpec := vmiInterfacesSpecByName[vmiIfaceStatus.Name]
+			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceSpec, vmiIfaceStatus, guestAgentInterface)
```

Comment:
Logically, everything looks fine.
However, one thing bothers me: The logic went to "deep" and got mixed with a function single responsibility.

It would be nicer to finish it at this level, with something like this:
```go
if vmiInterfacesSpecByName[vmiIfaceStatus.Name].Masquerade != nil {
    guestAgentIface.IPs = filterOutLinkLocalAddresses(guestAgentIface.IPs)
}
updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface)
```

Another option is to move this to a dedicated helper which takes into account the binding and processes the report based on it.
However, this may be more relevant in the future in case additional logic exists on a per-binding condition.

Reply:
> Since `updateVMIIfaceStatusWithGuestAgentData` handles IP parsing, moving this logic here also requires moving the parsing logic.
> I'm unsure if it will be an improvement.

I do not know what parsing you refer to.
This specific processing is about both address families, therefore there is no need to do any special handling beyond the one performed in the filter* helper.

Perhaps you can explain what other logic is needed beyond what I posted above.